go.mod

```diff
@@ -4,21 +4,21 @@ go 1.22.0
 
 require (
 	code.gitea.io/sdk/gitea v0.16.0
-	github.com/cnoe-io/argocd-api v0.0.0-20240530220153-91a5bf06f21d
+	github.com/cnoe-io/argocd-api v0.0.0-20240827200806-45206bdc4c03
 	github.com/docker/docker v24.0.9+incompatible
 	github.com/go-git/go-billy/v5 v5.5.0
 	github.com/go-git/go-git/v5 v5.12.0
-	github.com/go-logr/logr v1.4.1
+	github.com/go-logr/logr v1.4.2
 	github.com/google/go-cmp v0.6.0
 	github.com/google/go-github/v61 v61.0.0
-	github.com/spf13/cobra v1.8.0
+	github.com/spf13/cobra v1.8.1
 	github.com/stretchr/testify v1.9.0
-	k8s.io/api v0.29.1
-	k8s.io/apiextensions-apiserver v0.29.1
-	k8s.io/apimachinery v0.29.1
-	k8s.io/client-go v0.29.1
-	k8s.io/klog/v2 v2.120.1
-	sigs.k8s.io/controller-runtime v0.17.1
+	k8s.io/api v0.31.0
+	k8s.io/apiextensions-apiserver v0.31.0
+	k8s.io/apimachinery v0.31.0
+	k8s.io/client-go v0.31.0
+	k8s.io/klog/v2 v2.130.1
+	sigs.k8s.io/controller-runtime v0.19.0
```

Comment:
I think we want v0.30, not v0.31.  Controller-runtime should be 18.5.

Reply:
@AryanSharma9917 Any update on this? It's been a while since we released idpbuilder version and we implemented some QOL changes so I'd like to release one soon. Let me know if you have any questions.

Reply:
Done..!

Reply:
@AryanSharma9917 Can you set the version to `0.30.0` here or `0.30.1` in the argocd api library? It's set as `0.30.0` in the argocd api here: https://github.com/cnoe-io/argocd-api/blob/main/go.mod. I'd really like them to be in sync.

Reply:
This PR is outdated as argocd `go.mod` points now to the version `k8s.io/api v0.30.5`